### PR TITLE
Inherit common SSplot Parameters 

### LINKED ISSUE #2: Create template functions
Create @template for plotting parameters (ylim, main.title, pheight, etc.) so they are not specified in function every time. See r4ss man-oxygen for examples.

### LINKED ISSUE #66: ssplots: refactor "par" parameter to "mar"/plot-margin parameter
SSplot functions `SSPlotModelcomp`, `SSplotEnsemble`, `SSplotHCXval`, `SSplotJABBAres`, `SSplotRetro`, and `SSplotRunsTest` use a `par` parameter to setup the plot margins (_mar_). 

The use of  `par` the parameter is redundant, and can override ssplot graphical parameters.  
 
Replacing "par" with a _plot margin_ parameter would resolve this.

### PR DESCRIPTION
**Pull request by efletcherPIFSC**
Created generic functions `SSplotGeneric` , `SSplotGenericLegend`, `SSplotGenericPar`, `SSplotGenericUncertianty` to re-use common parameter documentation that ss3diags SSplot functions, via use roxygen tag`@inheritParams`. Resolves #2. 

Implementation of `SSplotGenericPar` resolves #66.

In addition:
- Included markdown notation to SSplot roxygen code documentation
- Added links to other functions such as r4ss::SS_Summarize, and replaced "TRUE/FALSE" with link to base::Logical
- Values in `subplots`, `indexfleets`, and `punits` are listed in a bulleted list
- Typo fixes

### CODE REVIEW

**Comment by efletcherPIFSC** [comment]
@MOshima-PIFSC 
Is the item description listed in the `subplots` parameter accurate for: 
- `SSPlotModelComp`
- `SSplotEnsemble`
- `SSplotHCxval`
- `SSplotJABBAres`
- `SSplotRetro`
- `SSplotRunsTest`

File: R/SSplotEnsemble.R [hunk 1]
```diff
@@ -1,73 +1,42 @@
...
-#' Plots model ensembles and forecasts with uncertaity represented by MVLN or MCMC posteriors
+#' Plots model ensembles and forecasts with uncertainty represented by MVLN or
+#' MCMC posteriors
 #'
-#' @param kb SSdeltaMVLN $kb type output
-#' @param subplots option to "Bratio","Fvalue","SSB", "F", "Recr","Catch"
+#' @param kb `kb` type output created by `SSdeltaMVLN()`
+#' @param subplots vector to create subplots with these options:
+#' \itemize{
+#'  \item `"stock"` Fish Population
```
Comment (MOshima-PIFSC) [inline R/SSplotEnsemble.R:9]: I would say stock is 'Stock biomass relative to reference point (B/BMSY)' and harvest 'Harvest rate relative to reference point (F/FMSY)'

File: R/SSplotHCxval.R [hunk 2]
```diff
@@ -1,81 +1,63 @@
...
-#' r4ss function summaryoutput().  Either "all" or a vector of numbers indicating
-#' columns in summary tables.
-#' @param Season option to specify Season as an integer of value 1-4 - Default uses first available, i.e. usual Seas = 1
+#' @param retroSummary List created by [r4ss::SSsummarize()] or
+#' [ss3diags::SSretroComps()]
+#' @param subplots optional use of the following:
+#' \itemize{
+#'  \item `"cpue"` Index data
+#'  \item `"len"` Length composition data
+#'  \item `"age"` Age composition data (yet to be tested)
```
Comment (MOshima-PIFSC) [inline R/SSplotHCxval.R]: Age composition data has been tested and works, can remove the 'yet to be tested'

File: R/SSplotHCxval.R [hunk 3]
```diff
@@ -1,81 +1,63 @@
...
-#' @param subplots optional use of c("cpue","len","age"), yet to be tested for age.
-#' @param models Optional subset of the models described in
-#' r4ss function summaryoutput().  Either "all" or a vector of numbers indicating
-#' columns in summary tables.
-#' @param Season option to specify Season as an integer of value 1-4 - Default uses first available, i.e. usual Seas = 1
+#' @param retroSummary List created by [r4ss::SSsummarize()] or
+#' [ss3diags::SSretroComps()]
+#' @param subplots optional use of the following:
+#' \itemize{
+#'  \item `"cpue"` Index data
```
Comment (MOshima-PIFSC) [inline R/SSplotHCxval.R]: to be consistent with other functions, label cpue as Index of abundance data

File: R/SSplotHCxval.R [hunk 4]
```diff
@@ -1,81 +1,63 @@
...
+#'  \item `"len"` Length composition data
```
Comment (MOshima-PIFSC) [inline R/SSplotHCxval.R]: nit pick but for length and age comp add a dash between the words (Length-composition data and Age-composition data)

File: R/SSplotJABBAres.R [hunk 5]
```diff
@@ -1,72 +1,41 @@
...
-#' it will be the directory where the model was run.
-#' @param filenameprefix Additional text to append to PNG or PDF file names.
-#' It will be separated from default name by an underscore.
-#' @param par list of graphics parameter values passed to par() function
-#' @param verbose Report progress to R GUI?
+#' @param ss3rep Stock Synthesis output as read by [r4SS::SS_output()]
+#' @param subplots string of type of data to plot:
+#' \itemize{
+#'  \item `"cpue"` Index of abundance data
+#'  \item `"len"` Length-composition
```
Comment (MOshima-PIFSC) [inline R/SSplotJABBAres.R]: nit pick but add data after length-composition, age-composition, and generalized size-composition

File: R/SSplotModelcomp.R [hunk 6]
```diff
@@ -1,83 +1,62 @@
...
+#' @param summaryoutput List created by [r4ss::SSsummarize()]
 #' @param subplots option to "SSB","Bratio","Fvalue","Recruits","Index"
-#' @param brp option to set reference point c("msy","btargs")
-#' @param fmsy to specify Fvalue as F/Fmsy if so in starter file setting
+#' \itemize{
+#'  \item `"SSB"` Spawning Stock Biomass
+#'  \item `"Bratio"` B Ratio denominator
+#'  \item `"Fvalue"` Fishing Mortality
+#'  \item `"Recruits"` Recruits
+#'  \item `"Index"` Index
```
Comment (MOshima-PIFSC) [inline R/SSplotModelcomp.R]: Index of abundance

File: R/SSplotModelcomp.R [hunk 7]
```diff
@@ -1,83 +1,62 @@
...
-#' columns in summary tables.
+#' @param summaryoutput List created by [r4ss::SSsummarize()]
 #' @param subplots option to "SSB","Bratio","Fvalue","Recruits","Index"
-#' @param brp option to set reference point c("msy","btargs")
-#' @param fmsy to specify Fvalue as F/Fmsy if so in starter file setting
+#' \itemize{
+#'  \item `"SSB"` Spawning Stock Biomass
+#'  \item `"Bratio"` B Ratio denominator
+#'  \item `"Fvalue"` Fishing Mortality
+#'  \item `"Recruits"` Recruits
```
Comment (MOshima-PIFSC) [inline R/SSplotModelcomp.R]: Age-0 recruits

File: R/SSplotModelcomp.R [hunk 8]
```diff
@@ -1,83 +1,62 @@
...
-#' @param models Optional subset of the models described in
-#' r4ss function summaryoutput().  Either "all" or a vector of numbers indicating
-#' columns in summary tables.
+#' @param summaryoutput List created by [r4ss::SSsummarize()]
 #' @param subplots option to "SSB","Bratio","Fvalue","Recruits","Index"
-#' @param brp option to set reference point c("msy","btargs")
-#' @param fmsy to specify Fvalue as F/Fmsy if so in starter file setting
+#' \itemize{
+#'  \item `"SSB"` Spawning Stock Biomass
+#'  \item `"Bratio"` B Ratio denominator
```
Comment (MOshima-PIFSC) [inline R/SSplotModelcomp.R]: Stock biomass relative to biomass reference point

File: R/SSplotRunstest.R [hunk 9]
```diff
@@ -53,60 +56,50 @@ ssruns_sig3 <- function(x, type = NULL, mixing = "less") {
...
-#' @param type Type parameter passed to points (default 'o' overplots points on
-#' top of lines)
-#' @param legend Option to add a legend. TRUE by default.
-#' @param legendloc Location of legend. Either a string like "topleft" or a vector
+#' @param ss3rep Stock Synthesis output as read by [r4ss::SS_output()]
+#' @param mixing `c("less","greater","two.sided")`. Default less is checking
+#' for positive autocorrelation only
+#' @param subplots optional flag for:
+#' \itemize{
+#'  \item `"cpue"` Index data
```
Comment (MOshima-PIFSC) [inline R/SSplotRunstest.R]: Index of abundance data

File: R/SSplotRunstest.R [hunk 10]
```diff
@@ -53,60 +56,50 @@ ssruns_sig3 <- function(x, type = NULL, mixing = "less") {
...
+#'  \item `"len"` Length composition data
```
Comment (MOshima-PIFSC) [inline R/SSplotRunstest.R:69]: nit pick again, the dashes between age/length/size composition

File: R/SSplotRunstest.R [hunk 11]
```diff
@@ -53,60 +56,50 @@ ssruns_sig3 <- function(x, type = NULL, mixing = "less") {
...
+#'  \item `"size"` Generalized size composition data
+#'  \item `"age"` Age composition data
+#'  \item `"con"` Conditional age at length data
```
Comment (MOshima-PIFSC) [inline R/SSplotRunstest.R]: nit pick Conditional age-at-length data

File: R/SSplotRunstest.R [hunk 12]
```diff
@@ -398,15 +392,26 @@ SSplotRunstest <- function(ss3rep = ss3diags::simple,
...
+#' indices or mean age or length and the type of input data (cpue, length comp,
+#' age comp, size comp, or conditional age-at-length).
 #'
 #' @param ss3rep Stock Synthesis output as read by r4SS function SS_output
-#' @param mixing c("less","greater","two.sided"). Default less is checking for positive autocorrelation only
-#' @param quants optional use of c("cpue","len","age","con"), default uses CPUE.
-#' @param indexselect Vector of fleet numbers for each model for which to compare
+#' @param mixing `c("less","greater","two.sided")`. Default less is checking for
+#' positive autocorrelation only
+#' @param quants optional use of `c("cpue","len","age","con")`, default uses
```
Comment (MOshima-PIFSC) [inline R/SSplotRunstest.R:404]: quants can be broken up into a bulleted list like subplots.

File: R/SSplotRunstest.R [hunk 13]
```diff
@@ -398,15 +392,26 @@ SSplotRunstest <- function(ss3rep = ss3diags::simple,
...
+#' CPUE.
```
Comment (MOshima-PIFSC) [inline R/SSplotRunstest.R]: nit pick but should probably change CPUE to "cpue" for clarity

**Review by MOshima-PIFSC — CHANGES_REQUESTED** [review]
Hey @efletcherPIFSC I looked over the subplots descriptions. They look good, I just added a few comments/suggestions to them. Also, I noted it in SSplotRunstest.R but any of the functions that have a quants argument, the same type of bulleted list can be used for those as well.

**Comment by efletcherPIFSC** [comment]
@MOshima-PIFSC  Didn't resolve any git code conflicts. Hence the dirty git conflict code. Fixing that.

**Review by MOshima-PIFSC — APPROVED** [review]
all changes look good. I approve the merge

**Comment by efletcherPIFSC** [comment]
This also resolves #67, #70

**Pull request merged**